---
title: "Microbial Communities in Alaskan Permafrost: Assessing the Compatibility of Genomic Data"
author: "Nora Fried"
format: pdf
editor_options: 
  chunk_output_type: console
bibliography: references.bib
---

## Abstract

Arctic permafrost soils store vast amounts of organic carbon and are experiencing accelerated thaw under climate warming. This feedback loop has significant implications for global nutrient cycling. Microbial community composition and activity mediate the decomposition of previously frozen soil organic matter, releasing greenhouse gases such as CO$_2$ and CH$_4$, yet the relative influence of primary landscape drivers, such as glacial drift and sample depth, remains unclear. Here, I re-analyze soils from D.V. Bakke's master's thesis, assessing two new DNA datasets (RunA and RunB) produced by the re-sequencing of soil samples, with the aim of addressing prior methodological issues that limited D.V. Bakke's analysis. The primary question driving this analysis is whether or not the new DNA data from RunA and RunB can be reliably combined for downstream analyses. Data processing involved data cleaning, rarefaction to standardize sequencing depth, ordination and diversity analyses. This initial assessment exposed a variety of sampling issues within the dataset and provides guidance for next steps in analyzing Bakke's Alaskan permafrost soil samples.

## Introduction

The Arctic is warming nearly four times faster than the global average, driving rapid environmental change across this sensitive ecosystem (@rantanen2022). One of the most significant consequences of warming is the thawing of permafrost, perennially frozen soils that have accumulated large reservoirs of soil organic carbons over thousands of years. Although permafrost regions cover only \~15% of the global land surface, they store more than twice as much carbon as the atmosphere (@schuur2015, @schuur2022). As permafrost thaws, these carbon pools are destabilized, creating a positive feedback to global climate change through the release of greenhouse gases (GHGs).

Microbes are central to this process of organic carbon degredation: as previously frozen organic matter becomes available, soil microbes rapidly metabolize it, releasing carbon dioxide (CO$_2$) and methane (CH$_4$) into the atmosphere (@mack2004). The composition and activity of microbial communities therefore directly influence the rate and magnitude of permafrost carbon cycling. Yet, microbial diversity across Arctic landscapes is highly heterogeneous, and the factors driving this variation remain poorly understood. In particular, geographic environmental gradients such as glacial drift and sample depth are likely to shape microbial community composition and diversity, but their relative contributions have not been systematically evaluated (@bakke2024).

Previous work by former UNH master's student D.V. Bakke in Dr. Jessica Ernakovich's lab investigated microbial communities in Arctic soil using microbial DNA extractions but methodological limitations and sequencing failures in Bakke's original sample set prevented robust statistical conclusions. To address these sampling issues, Bakke's soils were re-sequenced in the fall of 2025 by the Ernakovich lab, generating two new DNA datasets (RunA and RunB). The primary goal of the present study is to evaluate these new datasets and determine whether data from the two sequencing runs can be reliably combined for downstream analyses. Specifically, I ask whether the distributions of samples produced by RunA and RunB are similar enough to justify combining them. The analysis completed thus far involved a laborious process of data quality assurance, standardization of sequencing depth via rarefaction, and visualization through ordination and diversity analyses. 

Building on these microbial datasets, the aim is to provide a foundation for future studies exploring the influence of glacial drift and sample depth on permafrost microbial communities in Alaska. By re-assessing Bakke's samples with improved data quality and analytical rigor, this work establishes a framework for investigating the drivers of permafrost microbial diversity and informs predictions of microbial-mediated carbon flux under a warming Arctic.

## Methods:

#### Sample collection and History

Permafrost and active layer soil samples were collected from three distinct study sites on three different glacial drift deposits near Toolik Field Station, Alaska, in August 2021 by a team from UNH. Site soil materials range in age from 11,500 years to 125,000 years. Samples evaluated in this analysis are sourced from the oldest sampling site. See map below created by Fernando Montano Lopez for sample locations and glacial drift informatio (retrieved from Bakke, 2024). ![Map.](map.png)

#### Data Pre-Processing

PCR sample products were sent to the UNH Hubbard Center for Genome Studies during the fall of 2025 and sequenced to 2x250 base pare pair-end sequencing using Illumina Novaseq 6000. Amplicon sequencing data received back from UNH Hubbard Center was fed through the Ernaklovich lab's data2 pipeline on UNH's Premise supercomputer, transforming amplicon results and assigning ASV taxonomy to individual samples using Silva db v138 (@callahan2016, @holland-moritz2023). Data2 pipeline outputs were then fed into R and RStudio software for all additional manipulation.

#### Data Processing in R and RStudio

Data were imported into RStudio for processing and analysis. After initial data cleaning, the first step assess whether any samples were present in both RunA and RunB, as overlapping samples would allow for direct statistical comparison between sequencing runs. Unfortunately, no overlapping samples were detected, meaning that RunA and RunB contain completely distinct sets of samples. This absence of sample overlap limits the comparative analyses that can be performed, as any statistical differences detected between runs cannot be disentangled from underlying sample variation. Consequently, analyses were conducted separately for each run when necessary, with visual and descriptive comparisons used to evaluate sequencing depth, diversity, and community composition.

Sequencing depth was examined for each run and the data were standardized through rarefaction to create fair comparisons of microbial richness and diversity across samples. Ordination analyses were then performed using Bray--Curtis dissimilarity metrics, visualized through Principal Coordinates Analysis (PCoA) and Non-metric Multidimensional Scaling (NMDS) to assess patterns of community similarity. PERMANOVA was also applied to test for differences in community composition between runs; however, statistical results should be interpreted with caution due to the absence of overlapping samples between RunA and RunB, meaning observed differences may reflect distinct sample sets rather than true differences in microbial community.

```{r}
#| include: false
#load packages and data
library(tidyverse) # loads dplyr, ggplot, and other helpful data processing tools
library(vegan) # for dissimilarity metrics and other ecological functions
library(ape) # ape has a nice function for computing pcoas that has a little more functionality than cmdscale() in base R stats package
library(ggrepel) # optional - nice to help keep labels on ggplots from running into each other
library(stringr)

RunA <- read.csv("Textseqtab_wTax_mctoolsr_RunA_7.29.24.csv", check.names = FALSE) #check.names = FALSE gets rid of 'X' before column name
RunB <- read.csv("seqtab_wTax_mctoolsr_RunB_9.16.25.csv", check.names = FALSE)

if (colnames(RunA)[1] == "") colnames(RunA)[1] <- "ASV" #add ASV to first column name
if (colnames(RunB)[1] == "") colnames(RunB)[1] <- "ASV" #add ASV to first column name

#Extract and save taxonomy and associated ASV
taxonomy_A <- RunA[, c("ASV", "taxonomy"), drop = FALSE]
taxonomy_B <- RunB[, c("ASV", "taxonomy"), drop = FALSE]
write.csv(taxonomy_A, "RunA_taxonomy.csv", row.names = FALSE)
write.csv(taxonomy_B, "RunB_taxonomy.csv", row.names = FALSE)
```

```{r}
#| include: false
#check for overlapping samples between runs
runA_ids <- colnames(RunA)
runB_ids <- colnames(RunB)

runA_ids_clean <- runA_ids %>%
  str_trim() %>% # remove leading/trailing spaces
  toupper() %>% # make all uppercase
  str_remove("_L\\d{3}$")#remove run suffixes like _L001

runB_ids_clean <- runB_ids %>%
  str_trim() %>%
  toupper() %>%
  str_remove("_L\\d{3}$")

overlap_ids <- intersect(runA_ids_clean, runB_ids_clean)

length(overlap_ids)  # number of overlapping samples (2)
overlap_ids         # see which samples overlap (only ASV and Taxonomy columns)

```

```{r}
#| include: false
#Format data tables
RunA2 <- RunA
rownames(RunA2) <- RunA2$ASV
RunA2$ASV <- NULL #makes ASVs bold row names

RunB2 <- RunB
rownames(RunB2) <- RunB2$ASV
RunB2$ASV <- NULL

RunA2_counts <- RunA2[, sapply(RunA2, is.numeric)] #removes taxonomy "character" row (all rows should be numeric data now)
RunB2_counts <- RunB2[, sapply(RunB2, is.numeric)] #removes taxonomy "character" row

RunA2_counts <- RunA2_counts[, !(colnames(RunA2_counts) %in% c("74_S140_L001", "53_S135_L001"))]#removes problematic samples

rownames(RunA2_counts) <- paste0(rownames(RunA2_counts), "_A")
rownames(RunB2_counts) <- paste0(rownames(RunB2_counts), "_B")

combined_counts <- bind_rows(
  data.frame(SampleTotCounts = colSums(RunA2_counts), Run = "Run A"),
  data.frame(SampleTotCounts = colSums(RunB2_counts), Run = "Run B"))

# Transform ASV table so species are columns and rows are samples - this is the default that vegan and most other non-microbial ecological packages expects
RunA2_t <- t(RunA2_counts) # this converts from dataframe into matrix format
RunB2_t <- t(RunB2_counts)

#rownames(RunA2_t) <- paste0(rownames(RunA2_t), "_A") #label with A
#rownames(RunB2_t) <- paste0(rownames(RunB2_t), "_B") #label with B

# Run A2_t as a df with SampleID column
RunA_samples <- as.data.frame(RunA2_t)
RunA_samples$SampleID <- rownames(RunA_samples)

RunB_samples <- as.data.frame(RunB2_t)
RunB_samples$SampleID <- rownames(RunB_samples)

A_long <- RunA2_t %>%
  as.data.frame() %>%
  rownames_to_column("SampleID") %>%
  pivot_longer(-SampleID, names_to = "ASV", values_to = "count") %>%
  mutate(Run = "A")

B_long <- RunB2_t %>%
  as.data.frame() %>%
  rownames_to_column("SampleID") %>%
  pivot_longer(-SampleID, names_to = "ASV", values_to = "count") %>%
  mutate(Run = "B")

combined_long <- bind_rows(A_long, B_long)
combined_wide <- combined_long %>%
  pivot_wider(names_from = ASV, values_from = count, values_fill = 0)

combined_numeric <- combined_wide %>%
  column_to_rownames("SampleID") %>%
  select(where(is.numeric))

```

## Results and Discussion

Examining the distribution of sequencing depth, RunA and RunB exhibited substantial differences. The majority of samples in RunA had fewer than 50,000 reads (Figure 1), whereas RunB samples generally had much higher read counts, often hundreds of thousands (Figure 2). These differences in read depth are important to consider because they influence diversity analyses and ordination results. See RunA (above) and RunB (below) data distribution summary below highlighting these differences.

```{r fig.cap="RunA DNA reads distribution depth, majority of samples have less than 50000 reads.", fig.align='center', out.width='100%'}
#| message: false
#| echo: false
# Figure out rarefaction level sample A
summary(colSums(RunA2_counts)) #lots of very low reads, rarify to 1000

data.frame(SampleTotCounts = colSums(RunA2_counts)) %>%
  ggplot(aes(x = SampleTotCounts)) +
  geom_histogram(fill = "#FFBF29", color = "black") +
  theme_bw() +
  labs(
    x = "Total Sample Reads",
    y = "Count", 
    title = "Read Depth Distribution Run A")
```

```{r fig.cap="RunB DNA reads distribution depth, RunB is much more deeply sampled than RunA with many samples having hundreds of thousands of reads.", fig.align='center', out.width='100%'}
#| message: false
#| echo: false
# Figure out rarefaction level sample B
summary(colSums(RunB2_counts)) #lots of very low reads, rarify to 500 and loose alot of samples 

data.frame(SampleTotCounts = colSums(RunB2_counts)) %>%
  ggplot(aes(x = SampleTotCounts)) +
  geom_histogram(fill = "#14B49B", color = "black") +
  scale_x_continuous(labels = scales::comma) +
  theme_bw() +
  labs(
    x = "Total Sample Reads",
    y = "Count", 
    title = "Read Depth Distribution Run B")
```

```{r}
#| include: false
#What percentage of samples are have below 1000 reads? 
pct_below_1000 <- combined_counts %>%
  summarize(pct_below_1000 = mean(SampleTotCounts < 1000) * 100)

#pct_below_1000_a&b <- combined_counts %>%
  #group_by(Run) %>% 
  #summarize(pct_below_1000 = mean(SampleTotCounts < 1000) * 100)
```

Comparing raw sequencing counts between two different sequencing runs directly is inherently biased, as samples with higher read counts (ie. RunB) will appear to contain an increased diversity of microbial taxa. To enable fair comparisons across RunA and RunB, normalization is required; in this study, this was accomplished through rarefaction. Rarefaction involves selecting a target read depth (in this case, 1,000 reads per sample) and randomly subsampling each sample to that depth. Samples with fewer reads than the target are typically excluded, in this case 8.56% of samples are dropped during rarefaction. This subsampling procedure results in a dataset in which all retained samples have the same total read count, allowing unbiased comparisons of microbial richness, diversity, and community composition. Although a rarefaction depth of 1,000 reads is lower than what is typically recommended (5000 reads is often considered a minimum), it was chosen here to maximize the number of samples retained. Since the primary goal of this analysis is to determine whether the datasets from RunA and RunB can be combined, prioritizing sample retention over capturing rare taxa was appropriate for this pre-analysis. See Figure 3 for combined sample reads with rarefaction level.

To examine patterns of microbial community composition across samples, I calculated dissimilarities using the Bray--Curtis index, a metric that quantifies differences in community composition based on the relative abundance of taxa. These dissimilarity matrices were then visualized using two ordination methods: Principal Coordinates Analysis (PCoA) (Figure 4) and Non-metric Multidimensional Scaling (NMDS) (Figure 5). PCoA provides a metric-based ordination that preserves the actual distances between samples (as much as is possible in a reduced number of dimensions) while NMDS is a rank-based approach preserving the rank order of dissimilarities. Together, these ordination techniques allow for the visualization of community similarities and differences among samples, helping to identify potential clustering by sequencing run, sample depth, or other environmental factors.

Both the PCoA and NMDS analyses visually suggest substantial differences between the microbial communities represented in RunA and RunB. Unfortunately, statistical comparison of these runs is fundamentally limited because, again there are no overlapping samples between the runs. Despite this limitation, I conducted a PERMANOVA in an exploratory capacity, with the goal of determining whether the analysis could offer any additional insight into the data or help guide next steps for evaluating the two sequencing runs. PERMANOVA results below. 

```{r fig.cap="Combined sequencing depth with rarefaction level denoted at 1000 reads.", fig.align='center', out.width='100%'}
#| message: false
#| echo: false
#combined A and B plots
rar_level = 1000

ggplot(combined_counts, aes(x = SampleTotCounts, fill = Run)) +
  geom_histogram(position = "identity", bins = 30, color = "black") +
  scale_fill_manual(values = c("Run A" = "#FFBF29", "Run B" = "#14B49B")) +
  geom_vline(xintercept = rar_level, color = "red") +
  annotate("text",
         x = rar_level,
         y = max(table(cut(combined_counts$SampleTotCounts, 30))) * 1.05,
         label = "1000 reads",
         color = "red",
         angle = 90,
         hjust = 1,
         vjust = 1.25) +
  scale_x_continuous(labels = scales::comma) +
  theme_bw() +
  labs(
    x = "Total Sample Reads",
    y = "Count",
    title = "Read Depth Distribution Run A & B")
```

```{r}
#| include: false
# Rarefy data
combined_rar <- rrarefy(combined_numeric, rar_level)

rowSums(combined_rar) # check rarefying worked; all should have same read count = rar_level
```

```{r}
#| include: false
#Compute Bray-curtis dissimilarities distance matrices 

combined_bc <- vegdist(x = combined_rar, method = "bray")

```

```{r}
#| include: false
# PCOA ordination, with bray-curtis
combined_pcoa <- pcoa(combined_bc) # this is from the package ape; 
combined_pct <- round((combined_pcoa$values$Relative_eig) * 100, 1) # this line computes the variation on each axis

# NMDS oridination with bray-curtis
combined_nmds <-  metaMDS(combined_bc, k = 2, trymax = 100) # again, k = # of dimensions - though the impact of choosing k is slightly different in pcoa vs. nmds 
combined_nmds <-  metaMDS(combined_bc, k = 2, trymax = 100) 

#PCoA table
combined_pcoa_df <- combined_pcoa$vectors[,1:2] %>%
  data.frame() %>%
  rownames_to_column("SampleID") %>%
  dplyr::rename(PCOA1_BC = Axis.1, PCOA2_BC = Axis.2)

# NMDS plotting table
combined_nmds_df <- combined_nmds$points %>%
  data.frame() %>%
  rownames_to_column("SampleID") %>%
  dplyr::rename(MDS1_BC = MDS1, MDS2_BC = MDS2)

```

```{r}
#| include: false
#Join the tables together
combined_plot_df <- combined_wide %>%
  left_join(combined_nmds_df, by = "SampleID") %>%
  left_join(combined_pcoa_df, by = "SampleID")

```

```{r fig.cap="PCoA of Combined Samples.", fig.align='center', out.width='100%'}
#| message: false
#| echo: false
ggplot(combined_plot_df, aes(x = PCOA1_BC, y = PCOA2_BC, color = Run)) +
  geom_point(size = 3) +
  # geom_text_repel(aes(label = SampleID), 
  #                 size = 3,   
  #                 max.overlaps = 8, 
  #                 box.padding = 0.5, 
  #                 point.padding = 0.5) +
  scale_color_manual(values = c("A" = "#FFBF29", "B" = "#14B49B")) +
  theme_bw() +
  labs(x = "PCoA1", y = "PCoA2")
```

```{r fig.cap="NMDS of Combined Samples.", fig.align='center', out.width='100%'}
#| message: false
#| echo: false
ggplot(combined_plot_df, aes(x = MDS1_BC, y = MDS2_BC, color = Run)) +
  geom_point(size = 3) +
  # geom_text_repel(aes(label = SampleID), 
  #                 size = 3,   
  #                 max.overlaps = 8, 
  #                 box.padding = 0.5, 
  #                 point.padding = 0.5) +
  scale_color_manual(values = c("A" = "#FFBF29", "B" = "#14B49B")) +
  theme_bw() +
  labs(x = "NMDS1", y = "NMDS2")
```

```{r}
#| message: false
#| echo: false
adonis2(combined_bc ~ Run, data = combined_plot_df)

bd <- betadisper(combined_bc, combined_plot_df$Run) #betadispersion
anova(bd)
```

Results from the PERMANOVA suggest that RunA and RunB show a statistically significant but weak difference in community composition (PERMANOVA RÂ² = 0.075, p = 0.001), and betadispersion confirmed that within-group variances do not differ (p = 0.256). However, because the two runs contain completely distinct sets of samples, this result remains inconclusive - further emphasizing that that overlapping samples are the key to determining if these two data runs can be combined.

## Conclusions

The primary goal of this analysis was to evaluate two newly generated sequencing datasets (RunA and RunB) derived from the re-sequencing of soil samples originally analyzed in D.V. Bakke's master's thesis @bakke2024. Through examination of sample read-depth distribution, rarefaction, and plotted ordinations, this work attempts to address whether the two runs could be reliably combined for downstream ecological analyses. The results suggest clear visual differences between RunA and RunB in sequencing depth and community structure; however, because the runs share no overlapping samples, it is impossible to distinguish whether these differences reflect true biological variation or variation elsewhere along the sampling pipeline. The current evidence suggests that the two datasets should not be combined.

Moving forward, the most critical next step is generating overlapping samples across sequencing runs so that said variation can be statiscally evaluated. Establishing sequencing overlap will determine whether RunA and RunB can ultimately be merged into a unified dataset suitable for re-analyzing Bakke's original ecological hypotheses related to glacial drift and soil depth. In the meantime, each run can be futher analyzed independently to investigate within-run ecological patterns, including Shannon and Simpson diversity, which can provide additional insight into microbial community structure and variability within each dataset.
\newpage

